File: sdk/kotlin/examples/httpclient.kt
/**
 * HTTP client example — using a pre-configured Ktor HttpClient with optional proxy.
 *
 * Demonstrates the secondary constructor that accepts an HttpClient.
 *
 * Usage:
 *
 *   PACHCA_TOKEN=your_token PACHCA_CHAT_ID=12345 kotlin httpclient.kt
 */
package examples.httpclient

import com.pachca.sdk.PachcaClient
import com.pachca.sdk.PACHCA_API_URL
import io.ktor.client.*
import io.ktor.client.engine.*
import io.ktor.client.engine.cio.*
import io.ktor.http.Url
import io.ktor.client.plugins.*
import io.ktor.client.plugins.contentnegotiation.*
import io.ktor.serialization.kotlinx.json.*
import kotlinx.coroutines.runBlocking
import kotlinx.serialization.json.Json

fun main() = runBlocking {
    val token = System.getenv("PACHCA_TOKEN")
        ?: error("Set PACHCA_TOKEN environment variable")
    val chatId = System.getenv("PACHCA_CHAT_ID")?.toIntOrNull()
        ?: error("Set PACHCA_CHAT_ID environment variable")

    val http = HttpClient(CIO) {
        engine {
            val proxyUrl = System.getenv("HTTP_PROXY")
            if (proxyUrl != null) {
                proxy = ProxyBuilder.http(Url(proxyUrl))
            }
        }
        defaultRequest {
            headers.append("Authorization", "Bearer $token")
        }
        install(ContentNegotiation) {
            json(Json { ignoreUnknownKeys = true })
        }
    }

    val client = PachcaClient(client = http, baseUrl = PACHCA_API_URL)

    val chat = client.chats.getChat(chatId)
    println("Chat: ${chat.name} (id=${chat.id})")

    client.close()
}

File: sdk/python/examples/httpclient.py
"""
HTTP client example — using a pre-configured httpx.AsyncClient with optional proxy.

Demonstrates PachcaClient.from_client() with custom httpx client.

Usage:
    PACHCA_TOKEN=your_token PACHCA_CHAT_ID=12345 python examples/httpclient.py
    HTTP_PROXY=http://proxy:8080 PACHCA_TOKEN=... PACHCA_CHAT_ID=... python examples/httpclient.py
"""

import asyncio
import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "generated"))

import httpx
from pachca.client import PachcaClient, PACHCA_API_URL


async def main():
    token = os.environ["PACHCA_TOKEN"]
    chat_id = int(os.environ["PACHCA_CHAT_ID"])

    http = httpx.AsyncClient(
        base_url=PACHCA_API_URL,
        headers={"Authorization": f"Bearer {token}"},
        proxy=os.environ.get("HTTP_PROXY"),
    )
    client = PachcaClient.from_client(http)

    chat = await client.chats.get_chat(chat_id)
    print(f"Chat: {chat.name} (id={chat.id})")

    await client.close()


asyncio.run(main())
